core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/DefaultCodecRegistry.java

```diff
@@ -159,7 +159,13 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
-      return Objects.hash(cqlType, javaType, isJavaCovariant);
+      // NOTE: inlined Objects.hash for performance reasons (avoid Object[] allocation
+      // seen in profiler allocation traces)
+      int h1 = Objects.hashCode(cqlType);
+      int h2 = Objects.hashCode(javaType);
+      int h3 = Boolean.hashCode(isJavaCovariant);
+
+      return ((31 + h1) * 31 + h2) * 31 + h3;
```

Comment:
I don't _love_ the idea of actually having the multiplication logic inline like this... but I'm having trouble imagining how you'd get around it (for the reasons @tatu-at-datastax has already laid out).

Reply:
I haven’t seen the profile to know if this is called repeatedly on the same object, but given the final fields, is memoizing this even easier?

Reply:
Hmm, a thought does strike me though... can we even do away with the temporary ints?

```java
    private int hashFunction(int cqlTypeHash, int javaTypeHash, int isCovariantHash) {
      return ((31 + cqlTypeHash) * 31 + javaTypeHash) * 31 + isCovariantHash;
    }

    @Override
    public int hashCode() {
      return hashFunction(
          Objects.hashCode(cqlType), Objects.hashCode(javaType), Objects.hashCode(isJavaCovariant));
    }
```

Reply:
Agreed @tatu-at-datastax.  I considered something very much like that but I was afraid the inlining might make it less readable.  Looking at it now I agree we should be fine.

Reply:
@tatu-at-datastax I think I'd argue for the following:

```java
    public int hashCode() {
      // NOTE: inlined Objects.hash for performance reasons (avoid Object[] allocation
      // seen in profiler allocation traces)
      return ((31 + Objects.hashCode(cqlType)) * 31 + Objects.hashCode(javaType)) * 31
          + Boolean.hashCode(isJavaCovariant);
    }
```

It's your suggestion above + formatting changes based on Coveo.

Does that work for you?

Reply:
@absurdfarce It does. Changed & squashed.